packages/block-editor/src/components/rich-text/index.js

```diff
@@ -93,14 +95,18 @@ class RichTextWrapper extends Component {
 
 		if ( multiline ) {
 			if ( shiftKey ) {
-				onChange( insert( value, '\n' ) );
+				if ( ! disableLineBreaks ) {
+					onChange( insert( value, '\n' ) );
+				}
 			} else if ( canSplit && isEmptyLine( value ) ) {
 				this.onSplit( value );
 			} else {
 				onChange( insertLineSeparator( value ) );
 			}
 		} else if ( shiftKey || ! canSplit ) {
-			onChange( insert( value, '\n' ) );
+			if ( ! disableLineBreaks ) {
+				onChange( insert( value, '\n' ) );
+			}
```

Comment:
While reading the changes here, I was wondering why these onEnter handlers are block-editor specific and not something for the regular RichText component?

Reply:
Inserting line breaks on enter could be a default feature of `RichText` (in the rich text package), sure. But it should also be overridable. Right now there's just no default handling. Maybe in a future PR.